spark/v4.0/spark/src/main/java/org/apache/iceberg/spark/SparkSchemaUtil.java

```diff
@@ -199,7 +199,9 @@ public static Schema convert(Schema baseSchema, StructType sparkType, boolean ca
    * @param sparkType a Spark StructType
    * @return the equivalent Schema
    * @throws IllegalArgumentException if the type cannot be converted or there are missing ids
+   * @deprecated since 1.11.0, will be removed in 2.0.0
```

Comment:
all of these should say that these are going to be removed in 1.12.0 instead of 2.0.0, as the typical deprecation cycle is one minor version for non-iceberg-api modules.